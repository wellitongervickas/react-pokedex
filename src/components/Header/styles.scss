@import '../../assets/scss/variables';

.header {
  height: $size-big4 * 2;

  &, .header-navigation {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  .header-pokedex {
    display: flex;
    align-items: center;
  }

  .header-pokedex-ball, .header-pokedex-ball-2, .header-pokedex-ball-3 {
    border: $size-small4 / 2 solid $black;
  }

  .header-pokedex-ball {
    height: $size-big4;
    width: $size-big4;
    background-color: $water;
    border-radius: 100%;
  }

  .header-pokedex-ball-2, .header-pokedex-ball-3 {
    height: $size-default;
    width: $size-default;
    border-radius: 100%;
    margin-left: $size-small4;
  }

  .header-pokedex-ball-2 {
    background-color: $red;
  }

  .header-pokedex-ball-3 {
    background-color: $electric;
  }

  .header-navigation {
    cursor: pointer;
    color: $black;
  }

  .header-navigation > span {
    margin-left: $size-default;
  }
}
