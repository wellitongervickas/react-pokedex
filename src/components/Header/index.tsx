import React, {
  useCallback,
} from 'react';

import {
  useHistory,
} from 'react-router-dom';

import 'components/Header/styles.scss';

import { FaArrowLeft } from 'react-icons/fa';

export interface HeaderInterface {
  useNavigation: boolean,
}

const Header = ({ useNavigation }: HeaderInterface) => {
  const { push } = useHistory();

  const handleBack = useCallback(() => {
    push('/');
  }, [push])

  return (
    <div className="header">
      <div className="header-pokedex">
        <div className="header-pokedex-ball" />
        <div className="header-pokedex-ball-2" />
        <div className="header-pokedex-ball-3" />
      </div>
      {useNavigation && (
        <div
          className="header-navigation"
          onClick={handleBack}
        >
          <FaArrowLeft />
          <span>Go back</span>
        </div>
      )}
    </div>
  )
};

Header.defaultProps = {
  useNavigation: true,
};

export default Header;
