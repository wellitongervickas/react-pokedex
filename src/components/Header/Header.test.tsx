import React from 'react';
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import { render, fireEvent } from '@testing-library/react';

import '@testing-library/jest-dom/extend-expect'

import Header from '.';

describe('Component Header', () => {
  const history = createMemoryHistory();

  it('renders Header component', () => {
    const { container } = render(
      <Router history={history}>
        <Header />
      </Router>
    );

    const tree = container.firstElementChild;

    expect(tree?.classList.contains('header')).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });

  it('should renders balls', () => {
    const { container } = render(
      <Router history={history}>
        <Header />
      </Router>
    );

    const tree = container.firstElementChild?.firstElementChild;

    expect(tree?.classList.contains('header-pokedex')).toBeTruthy();
    expect(tree?.children[0].classList.contains('header-pokedex-ball')).toBeTruthy();
    expect(tree?.children[1].classList.contains('header-pokedex-ball-2')).toBeTruthy();
    expect(tree?.children[2].classList.contains('header-pokedex-ball-3')).toBeTruthy();

    expect(tree).toMatchSnapshot();
  });

  it('should render with use navigation', () => {
    const { container } = render(
      <Router history={history}>
        <Header useNavigation={true} />
      </Router>
    );

    const tree = container.firstElementChild?.children[1];

    expect(tree?.classList.contains('header-navigation')).toBeTruthy();
    expect(tree).toMatchSnapshot();

  });

  it('should render without use navigation', () => {
    const { container } = render(
      <Router history={history}>
        <Header useNavigation={false} />
      </Router>
    );

    const tree = container.firstElementChild?.children[1];

    expect(tree).toBeUndefined();
    expect(tree).toMatchSnapshot();
  });

  it('should handle event back', () => {
    history.push('/new-route');
    expect(history.location.pathname).toBe('/new-route');

    const { getByText } = render(
      <Router history={history}>
        <Header />
      </Router>
    );

    fireEvent.click(getByText(/Go back/i));

    expect(history.location.pathname).toBe('/');
  });
});
