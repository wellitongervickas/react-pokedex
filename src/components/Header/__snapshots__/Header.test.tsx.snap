// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Header renders Header component 1`] = `
<div
  class="header"
>
  <div
    class="header-pokedex"
  >
    <div
      class="header-pokedex-ball"
    />
    <div
      class="header-pokedex-ball-2"
    />
    <div
      class="header-pokedex-ball-3"
    />
  </div>
  <div
    class="header-navigation"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
      />
    </svg>
    <span>
      Go back
    </span>
  </div>
</div>
`;

exports[`Component Header should render with use navigation 1`] = `
<div
  class="header-navigation"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
    />
  </svg>
  <span>
    Go back
  </span>
</div>
`;

exports[`Component Header should render without use navigation 1`] = `undefined`;

exports[`Component Header should renders balls 1`] = `
<div
  class="header-pokedex"
>
  <div
    class="header-pokedex-ball"
  />
  <div
    class="header-pokedex-ball-2"
  />
  <div
    class="header-pokedex-ball-3"
  />
</div>
`;
